src/utils/utilities.ts

```diff
@@ -240,3 +244,189 @@ export const calculateTimeLeft = (createdAt: string) => {
   // @ts-ignore
   return Math.max(0, Math.floor((targetTime - currentTime) / 1000));
 };
+
+export const generateDataFromPSBT = (base64Str: string, signer: Signer) => {
+  try {
+    const psbt = bitcoin.Psbt.fromBase64(base64Str);
+
+    // Convert the buffer to a hex string and reverse it to get the transaction ID
+    const txId = Buffer.from(psbt.txInputs[0].hash).reverse().toString('hex');
+    const vout = psbt.txInputs[0].index;
+
+    const changeAddress = getChangeAddress(base64Str);
+    const vBytes = estimateVByteFromPSBT(base64Str);
+
+    const signersList = [];
+    let signerMatched = false;
+
+    psbt.data.inputs.forEach((input) => {
+      if (input.bip32Derivation) {
+        // Loop through all derivations (in case there are multiple keys)
+        input.bip32Derivation.forEach((derivation) => {
+          const data = {
+            derivationPath: shortDerivationPath(derivation.path),
+            masterFingerprint: derivation.masterFingerprint.toString('hex'),
+            xpriv: '',
+            xpub: derivation.pubkey.toString('hex'),
+          };
+          signersList.push(data);
+          if (data.masterFingerprint.toLowerCase() === signer.masterFingerprint.toLowerCase()) {
```

Comment:
should make sure the derivation path matches as well for one of the signer's keys

Reply:
added validation for pubKey